src/api_manager/service_control/aggregated.cc

```diff
@@ -305,6 +305,16 @@ Status Aggregated::Close() {
   return Status::OK;
 }
 
+void Aggregated::SendEmptyReport() {
+  ReportRequest request;
+  ReportResponse* response = new ReportResponse;
+  Call(request, response,
+       [this, response](const ::google::protobuf::util::Status&) {
+         delete response;
```

Comment:
I'm confused: Where does ESP read the latest rollout ID from the `ReportResponse`? Shouldn't it be using the `response` to check for it?

Reply:
Just using the existing code to process the Check/Report response.  It is in the HandleRespones() function.

Reply:
This change only adds timer to trigger a dummy Report calls in every minutes.  The existing code will handle the rest:   on every check/response code it will check rollout id,  then it set to a callback function which is intercepted by config_manager.